src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
+static std::map<std::string, struct curl_context> contexts;
+
+struct curl_context
```

Comment:
We don't follow the style guide to 100%
The easiest way is probably to just look at the code and follow its example.
Unfortunately not all the code is up-to-date, for example part of mongoBackend uses 4 spaces for indentation. Some day I might have time to fix this ... [ And add a make step to assure that the code follows the style guide - the more implementors we are, the more important it gets ]